chromium_src/components/permissions/permission_request_manager.cc

```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <string>
+
+#include "brave/components/brave_wallet/common/buildflags/buildflags.h"
+#include "components/permissions/permission_request.h"
+
+#if BUILDFLAG(BRAVE_WALLET_ENABLED)
+#include "brave/components/brave_wallet/browser/ethereum_permission_utils.h"
+
+namespace permissions {
+namespace {
+bool BraveShouldGroupRequests(PermissionRequest* a, PermissionRequest* b);
+}  // namespace
+}  // namespace permissions
+
+#define BRAVE_SHOULD_GROUP_REQUESTS   \
+  if (BraveShouldGroupRequests(a, b)) \
+    return true;
+#else
+#define BRAVE_SHOULD_GROUP_REQUESTS
+#endif  // BUILDFLAG(BRAVE_WALLET_ENABLED)
+
+#include "../../../../components/permissions/permission_request_manager.cc"
+#undef BRAVE_SHOULD_GROUP_REQUESTS
+
+#if BUILDFLAG(BRAVE_WALLET_ENABLED)
+namespace permissions {
+
+namespace {
+
+bool BraveShouldGroupRequests(PermissionRequest* a, PermissionRequest* b) {
+  std::string origin_a;
+  std::string origin_b;
+  if (a->GetRequestType() == RequestType::kBraveEthereum &&
+      b->GetRequestType() == RequestType::kBraveEthereum &&
+      brave_wallet::ParseRequestingOrigin(a->GetOrigin(), true, &origin_a,
+                                          nullptr) &&
+      brave_wallet::ParseRequestingOrigin(b->GetOrigin(), true, &origin_b,
+                                          nullptr) &&
+      origin_a == origin_b) {
+    return true;
+  }
+
+  return false;
+}
+
+bool IsAccepted(PermissionRequest* request,
+                const std::vector<std::string>& accounts) {
+  for (auto account : accounts) {
+    if (base::EndsWith(request->GetOrigin().host(), account,
+                       base::CompareCase::INSENSITIVE_ASCII)) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+}  // namespace
+
+void PermissionRequestManager::AcceptEthereumPermissionRequests(
+    const std::vector<std::string>& accounts) {
+  std::vector<PermissionRequest*>::iterator requests_iter;
+  for (requests_iter = requests_.begin(); requests_iter != requests_.end();
+       requests_iter++) {
+    DCHECK((*requests_iter)->GetRequestType() == RequestType::kBraveEthereum);
+    if (IsAccepted(*requests_iter, accounts)) {
+      PermissionGrantedIncludingDuplicates(*requests_iter,
+                                           /*is_one_time=*/false);
+    } else {
+      CancelledIncludingDuplicates(*requests_iter);
+    }
+  }
+
+  // TODO(jocelyn): This does not have any bad impact atm if we finalize all
+  // requests with GRANTED option in the situation that some sub-requests are
+  // not granted, but we need to take a deeper look to see how we can finalize
+  // requests with different actionos.
```

Comment:
actions

Reply:
Fixed in https://github.com/brave/brave-core/pull/9443/commits/dfffa806c1357c7e7d7e1736dfc9da3e33f98bd3